packages/SwingSet/src/kernel/messageResult.js

```diff
@@ -0,0 +1,68 @@
+/* global harden */
+
+import { assert, details } from '@agoric/assert';
+import { insistCapData } from '../capdata';
+
+export function makeMessageResult(message, resultPolicy, panic) {
+  let currentStatus = 'pending';
+  let eventualResolution;
+
+  assert(
+    resultPolicy === 'logAlways' ||
+      resultPolicy === 'logFailure' ||
+      resultPolicy === 'panic' ||
+      resultPolicy === 'ignore',
+    details`invalid result policy ${resultPolicy}`,
+  );
+  assert.typeof(panic, 'function');
+
+  function log() {
+    // prettier-ignore
+    console.log(
+      `kernel message ${message}: ${currentStatus} ${JSON.stringify(eventualResolution)}`,
+    );
+  }
+
+  return harden({
+    status() {
+      return currentStatus;
+    },
+    resolution() {
+      if (currentStatus === 'pending') {
+        throw new Error(`resolution of result is still pending`);
+      } else {
+        return eventualResolution;
+      }
+    },
+    noteResolution(newStatus, resolvedTo) {
```

Comment:
How hard would it be to split the write-side and read-side facets of this object? The exogenous `queueToExport` caller should probably not get the ability to resolve this object, only the ability to sense the outcome.

Also we should `harden()` everything this returns, as a matter of general principle.

Reply:
I didn't originally bother on the theory that the controller's client (who is the one who gets it) has god power anyway, but splitting it lets that client hand it to somebody less trusted.  Trivial to do.  Also, stuff already was hardened.